### PR TITLE
Check audience no protocol scheme

### PR DESCRIPTION
**Pull request by JimmyCYJ**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #https://github.com/istio/proxy/issues/991

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: src/envoy/http/jwt_auth/pubkey_cache.h [hunk 1]
```diff
@@ -83,6 +98,25 @@ class PubkeyCacheItem {
   }
 
  private:
+  // Searches "http://" or "https://" prefix in aud, and stores the position
+  // after protocol scheme prefix in beg. Searches trailing slash in aud, and
+  // stores the position before trailing slash in end. Returns true if aud has
+  // characters other than prefix and suffix.
+  bool SanitizeAudience(const std::string& aud, int& beg, int& end) {
```
Comment (qiwzhang) [inline src/envoy/http/jwt_auth/pubkey_cache.h]: I think it is better for this function to return a processed string.
If not modification, (for most case), just return the input string which is not data copy (as I understand std::string) which is fast.

File: src/envoy/http/jwt_auth/pubkey_cache.h [hunk 2]
```diff
@@ -27,6 +27,10 @@ namespace JwtAuth {
 namespace {
 // Default cache expiration time in 5 minutes.
 const int kPubkeyCacheExpirationSec = 600;
+
+const std::string kHTTPSchemePrefix("http://");
```
Comment (qiwzhang) [inline src/envoy/http/jwt_auth/pubkey_cache.h]: add comment per style guide for each global variable

Comment (JimmyCYJ) [reply src/envoy/http/jwt_auth/pubkey_cache.h]: Done.

File: src/envoy/http/jwt_auth/pubkey_cache.h (hunk 1 above)
Comment (JimmyCYJ) [reply src/envoy/http/jwt_auth/pubkey_cache.h]: Done.

**Comment by JimmyCYJ** [comment]
Please take a look. Thanks!

**Pull request merged**